authorize_request_handler.go

```diff
@@ -104,23 +109,23 @@ func (f *Fosite) authorizeRequestParametersFromOpenIDConnectRequest(request *Aut
 		case *jwt.SigningMethodRSA:
 			key, err := f.findClientPublicJWK(oidcClient, t, true)
 			if err != nil {
-				return nil, errors.WithStack(ErrInvalidRequestObject.WithHint("Unable to retrieve RSA signing key from OAuth 2.0 Client.").WithCause(err).WithDebug(err.Error()))
+				return nil, errors.WithStack(ErrInvalidRequestObject.WithHintf("Unable to retrieve RSA signing key from OAuth 2.0 Client because: %s.", err).WithCause(err).WithDebug(err.Error()))
```

Comment:
No, we should never leak `err` out, just when debugging. I removed this recently in one PR. So `because ...` bit I worry about. (Especially if we will implement custom key search strategies with https://github.com/ory/fosite/issues/452)

Reply:
Hm I see your point but the problem is that this is really hard to debug. You would have to turn on debug mode for clients to understand why their request failed (e.g. `kid` does not exist, wrong alg, ...). So we need a way to return the appropriate error to the regular OAuth2 consumer without enabling debug.

Reply:
I think we should then check for known error messages and map it? Like [it is done here](https://github.com/ory/fosite/blob/master/handler/oauth2/strategy_jwt.go#L108-L134) already?